### PR TITLE
sjawhar-legion-523: keep repo clones fresh — fetch before dispatch, update on close

### LINKED ISSUE #523: feat(daemon): keep default repo clone updated — fetch before dispatch, update on issue close
## Problem

The daemon maintains a default clone per repo via `ensureRepoClone`, but it only clones once and does a background fetch. Creating workspaces from a stale clone means workers start with outdated code and waste time fetching.

## Current behavior
- `ensureRepoClone` checks if clone exists, skips if so
- Background fetch runs but clone can be stale
- No update on issue close

## Desired behavior
1. **Before each implementer dispatch**: `jj git fetch` on the default clone (fast incremental fetch)
2. **On issue close**: Update the default clone to latest main (merge just landed)
3. **On daemon startup**: Fetch all tracked repo clones to warm them up

## Files
`packages/daemon/src/daemon/repo-manager.ts` — `ensureRepoClone` and `ensureWorkspace`

## Priority
High — directly impacts dispatch speed, especially for large repos like agent-c.

### PR DESCRIPTION
**Pull request by sjawhar**
Closes #523

## Summary

Keep default repo clones fresh so workers start with up-to-date code instead of fetching from scratch.

**Three changes in repo-manager.ts / server.ts / index.ts:**

1. **Blocking fetch before implementer dispatch** — `POST /workers` with `mode=implement` now runs a blocking `jj git fetch` on the default clone *before* `ensureWorkspace()` creates the jj workspace. Other modes retain the existing non-blocking background fetch after workspace creation. If the pre-dispatch fetch fails, it logs a warning and proceeds (non-fatal).

2. **Background fetch on issue close** — `cleanupDoneIssueWorkers()` now fires a non-blocking `startBackgroundFetch()` for each unique repo after workspace cleanup. Since Done means a merge just landed, this brings the clone up to date for future dispatches.

3. **Warm all clones on daemon startup** — new `fetchAllTrackedRepos()` function walks `reposDir/{host}/{owner}/{repo}` and fetches every clone. Called non-blocking during `startDaemon()` so it doesn't delay startup.

## Testing

- 5 new unit tests for `fetchAllTrackedRepos` (multi-repo, error collection, empty, no listDir, nonexistent)
- 3 new server tests (blocking fetch for implement, non-blocking for plan, post-close fetch)
- Updated existing test expectations for changed jj command ordering
- All 212 daemon tests pass, biome clean

### CODE REVIEW

**Comment by sjawhar** [comment]
## Review Summary

**CRITICAL (P1):** 0 issues
**IMPORTANT (P2):** 0 issues
**MINOR (P3):** 2 suggestions

**Verdict: Approved to merge.**

All three requirements from #523 are correctly implemented:

1. **Blocking fetch before implementer dispatch** — `server.ts:931-938` awaits `startBackgroundFetch` for `WorkerMode.IMPLEMENT`, with graceful fallback on failure. Non-implement modes retain the existing non-blocking fetch.
2. **Background fetch on issue close** — `server.ts:534-543` fires non-blocking `startBackgroundFetch` for each unique repo after Done issue cleanup. Deduplication via `cleanedRepoRefs` Map prevents redundant fetches.
3. **Warm all clones on daemon startup** — `index.ts:405-421` calls `fetchAllTrackedRepos()` non-blocking during `startDaemon()`, with proper error logging.

Error handling is solid throughout — all fetch operations are non-fatal and logged as warnings/errors. Edge cases (missing clone, auth failure, empty dirs, missing listDir dep) are handled and tested.

### Evidence
- CI status: lint pass, test pass (178 server tests, 34 repo-manager tests), typecheck pass
- Mergeable: MERGEABLE (no conflicts)
- No bot review comments to address
- Test coverage: 5 new unit tests for `fetchAllTrackedRepos`, 3 new server integration tests, 1 updated existing test with ordering assertion
- The ordering test at `server.test.ts:474` verifies fetch fires BEFORE clone and workspace add for implement mode

### P3 Findings
1. **Naming:** `startBackgroundFetch` is slightly misleading when `await`-ed for the implement mode blocking path. A name like `fetchRepoClone` would better describe the dual usage. Comments at the call site explain the intent clearly, so this is cosmetic.
2. **Sequential fetching:** `fetchAllTrackedRepos` fetches repos sequentially via nested for-loops. Parallel `Promise.allSettled` would be faster for many repos, but sequential avoids thundering herd issues on shared jj repos. This is a reasonable design choice for correctness.

File: packages/daemon/src/daemon/server.ts [hunk 1]
```diff
@@ -909,6 +925,18 @@ export function startServer(opts: ServerOptions): {
               if (!repoRef) {
                 return badRequest("invalid_repo: expected owner/repo");
               }
+              // For implement mode: blocking fetch BEFORE workspace creation so the
+              // workspace is created from a fresh clone. For other modes: non-blocking
+              // fetch after workspace creation (existing behavior).
+              if (mode === WorkerMode.IMPLEMENT) {
+                try {
+                  await startBackgroundFetch(opts.paths, repoRef, opts.repoManagerDeps);
```
Comment (sjawhar) [inline packages/daemon/src/daemon/server.ts:933]: **P3 (Minor):** The function name `startBackgroundFetch` is slightly misleading when called with `await` here for the blocking pre-dispatch path. A name like `fetchRepoClone` would more accurately describe the dual usage (blocking for implement, non-blocking for other modes). The comments here explain the intent well, so this is cosmetic only.

File: packages/daemon/src/daemon/repo-manager.ts [hunk 2]
```diff
@@ -177,6 +177,58 @@ export async function startBackgroundFetch(
...
+      const ownerPath = path.join(hostPath, owner);
+      const repos = await listDir(ownerPath);
+      for (const repo of repos) {
+        const clonePath = path.join(ownerPath, repo);
+        if (!(await deps.exists(clonePath))) {
+          continue;
+        }
+        const label = `${host}/${owner}/${repo}`;
+        try {
+          const result = await deps.runJj(["git", "fetch", "-R", clonePath]);
```
Comment (sjawhar) [inline packages/daemon/src/daemon/repo-manager.ts:213]: **P3 (Minor):** Fetches run sequentially here via nested for-loops. For a daemon tracking many repos, parallel fetching with `Promise.allSettled` would be faster. However, sequential is a reasonable choice — it avoids thundering herd on shared jj repos and keeps resource usage predictable during startup.
